tools/loader/ExecutorCore.cpp

```diff
@@ -304,25 +357,37 @@ int Executor::executeNetwork() {
     // streaming.
     bool isFirstRun = true;
 
+    // Perform graph profiling initialization if needed.
+    // if (profilingGraph()) {
+    //  loader.initGraphProfiling(
+    //      bindings, miniBatch > 0 ? miniBatch :
+    //      inputImageFilenames_[0].size(), inputImageFilenames_[0].size());
+    //}
+
     // These will be set during the first run.
-    Placeholder *inputImagePH = nullptr;
-    std::vector<Placeholder *> outputPHV;
-    llvm::StringMap<Placeholder *> PHM;
+    llvm::StringMap<Placeholder *> iPHM;
+    llvm::StringMap<Placeholder *> oPHM;
+    std::vector<Placeholder *> inPHs;
+    std::vector<Placeholder *> outPHs;
 
     size_t miniBatchIndex = startIndex;
-    Tensor inputImageData;
+    std::vector<Tensor> inputData(modelInputsOpt.size());
     if (preloadAllImages) {
-      inputImageData = preloadedInputImageData.getUnowned();
+      inputData[0] = preloadedInputImageData.getUnowned();
     }
-    std::vector<std::string> inputImageBatchFilenames;
-    if (!miniBatchMode && !streamInputFilenamesMode) {
-      inputImageBatchFilenames = inputImageFilenames;
+
+    DoubleVec<std::string> inputImageBatchFilenames;
+    if ((!miniBatchMode) && (!streamInputFilenamesMode)) {
+      inputImageBatchFilenames = inputImageFilenames_;
+    } else if (singleBatchRepeatedMode && !miniBatchMode) {
+      inputImageBatchFilenames = inputImageFilenames_;
```

Comment:
Seems like these two can be combined? E.g.
```suggestion
    if ((!miniBatchMode) && 
        (!streamInputFilenamesMode || singleBatchRepeatedMode)) {
      inputImageBatchFilenames = inputImageFilenames_;
```